### PR TITLE
#421 using Moo

### PR DESCRIPTION
**Pull request by ruzhnikov**
The first ball in the porting to Moo

### CODE REVIEW

File: share/shutter/resources/modules/Shutter/App/Common.pm [hunk 1]
```diff
@@ -22,419 +23,171 @@
 
 package Shutter::App::Common;
 
-#modules
-#--------------------------------------
 use utf8;
-use strict;
```
Comment (DarthGandalf) [inline share/shutter/resources/modules/Shutter/App/Common.pm:28]: Does Moo imply strict?

File: share/shutter/resources/modules/Shutter/App/Common.pm [hunk 2]
```diff
@@ -22,419 +23,171 @@
...
+    my $self = shift;
+
+    my ( $window_at_pointer, $x, $y, $mask ) = Gtk3::Gdk::get_default_root_window->get_pointer;
+    my $mon = Gtk3::Gdk::Screen::get_default->get_monitor_geometry(
+        Gtk3::Gdk::Screen::get_default->get_monitor_at_point( $x, $y ) );
+
+    return ($mon);
+}
+
+# Methods that used in old realization and needed for backward compatibility
```
Comment (DarthGandalf) [inline share/shutter/resources/modules/Shutter/App/Common.pm]: implementation :)

realization in English means something different than in Russian:
"I realized something" - "Я что-то понял"

File: share/shutter/resources/modules/Shutter/App/OldCommon.pm [hunk 3]
```diff
@@ -0,0 +1,447 @@
+###################################################
```
Comment (DarthGandalf) [inline share/shutter/resources/modules/Shutter/App/OldCommon.pm]: Is this still needed?

**Review by DarthGandalf — APPROVED** [review]
Thanks!

File: share/shutter/resources/modules/Shutter/App/OldCommon.pm (hunk 3 above)
Comment (ruzhnikov) [reply share/shutter/resources/modules/Shutter/App/OldCommon.pm]: Just in case. It can be removed in the future

Comment (DarthGandalf) [reply share/shutter/resources/modules/Shutter/App/OldCommon.pm]: If it needs to be restored, we have git history

File: share/shutter/resources/modules/Shutter/App/Common.pm (hunk 1 above)
Comment (ruzhnikov) [reply share/shutter/resources/modules/Shutter/App/Common.pm:28]: Yes
```bash
$ perl -le 'use Moo; $xxx = 10'
Global symbol "$xxx" requires explicit package name (did you forget to declare "my $xxx"?) at -e line
```

File: share/shutter/resources/modules/Shutter/App/Common.pm (hunk 2 above)
Comment (ruzhnikov) [reply share/shutter/resources/modules/Shutter/App/Common.pm]: Ok) next time will use "implementation"

Comment (DarthGandalf) [reply share/shutter/resources/modules/Shutter/App/Common.pm]: I mean, update the comment please

File: share/shutter/resources/modules/Shutter/App/OldCommon.pm (hunk 3 above)
Comment (ruzhnikov) [reply share/shutter/resources/modules/Shutter/App/OldCommon.pm]: done in 12d391c4

File: share/shutter/resources/modules/Shutter/App/Common.pm (hunk 2 above)
Comment (ruzhnikov) [reply share/shutter/resources/modules/Shutter/App/Common.pm]: done in 12d391c4

**Pull request merged**